astropy/coordinates/representation.py

```diff
@@ -538,12 +773,24 @@ def _combine_operation(self, op, other, reverse=False):
             Whether the operands should be reversed (e.g., as we got here via
             ``self.__rsub__`` because ``self`` is a subclass of ``other``).
         """
+        self._raise_if_has_differentials(op.__name__)
+
         result = self.to_cartesian()._combine_operation(op, other, reverse)
         if result is NotImplemented:
             return NotImplemented
         else:
             return self.from_cartesian(result)
 
+    # We need to override this setter to support differentials
+    @BaseRepresentationOrDifferential.shape.setter
+    def shape(self, shape):
+        # See: https://stackoverflow.com/questions/3336767/ for an example
+        BaseRepresentationOrDifferential.shape.fset(self, shape)
```

Comment:
Same shape is not good enough (see above), because unlike `reshape`, the shape setter errors if the reshape is not possible (it has to, since it is inplace). My suggestion is to do the shape-setting on the differentials in a `try/except` loop, and reverse it if it fails anywhere. See the normal `_apply`